### ISSUE TITLE
Mil Stdr 2525C Graphics Render Improperly

### ISSUE THREAD

**Issue by fteel** [missioncommand/mil-sym-android#96]
The following graphics do not render properly:
Taks->Disrupt
MOBILITY/SURVIVABILITY -> OBSTACLES -> OBSTACLE EFFECT -> DISRUPT  
Chemically Contaminated Area
Halted Convoy
Biologically Contaminated Area
Minimum Safe Distance Zones
Roadblock Complete (executed)
Dynamic Depiction
Dummy Minefield (dynamic)
Biologically Contaminated Area
Biologically Contaminated Area

Also, all graphics that indicate it as a rectangle render as a square.

See images in Attachments
![Slide1](https://user-images.githubusercontent.com/62301296/93824635-66656080-fc29-11ea-9d41-72b7a295fa30.PNG)
![Slide2](https://user-images.githubusercontent.com/62301296/93824641-682f2400-fc29-11ea-8e30-037e69261d1a.PNG)
![Slide3](https://user-images.githubusercontent.com/62301296/93824644-68c7ba80-fc29-11ea-8ad4-4ba920d5062b.PNG)
![Slide4](https://user-images.githubusercontent.com/62301296/93824647-69605100-fc29-11ea-84ea-db0667e25e32.PNG)
![Slide5](https://user-images.githubusercontent.com/62301296/93824652-69f8e780-fc29-11ea-94ee-ada4472e0af7.PNG)
![Slide6](https://user-images.githubusercontent.com/62301296/93824656-6a917e00-fc29-11ea-9edd-44c907071f89.PNG)

**Comment by michael-spinelli** [comment]
Let me ask a few things before I look deeper into this.  

-Going by the Disrupt example, it looks like you didn't change the symbology standard from 2525B to 2525C.
you can do this globally: 
[RendererSettings.getInstance().setSymbologyStandard(RendererSettings.Symbology_2525C); ](https://github.com/missioncommand/mil-sym-android/wiki#21configuring-the-renderer-for-your-needs)
or per graphic: 
[attributes.put(MilStdAttributes.SymbologyStandard, String.valueOf(RendererSettings.Symbology_2525Bch2_USAS_13_14));  ](https://github.com/missioncommand/mil-sym-android/wiki#322setting-attributes)

-Without knowing what the point placement is, the Disrupt looks like it's rendering ok.

-Contaminated areas, is your issue with the label placement?  This could be affected by the map scale value you're using.

-Looking at the minefields, I think you need to make sure you're setting the [map scale](https://github.com/missioncommand/mil-sym-android/wiki#3316scale-for-3d) properly.  Unfortunately, the "right" value varies depending on the map technology.  

-For the area's, are you passing in the appropriate width values?

[Multi Point Tactical Graphics:  ](https://github.com/missioncommand/mil-sym-android/wiki#321setting-modifiers)
//(comma delimited for modifiers with multiple values)
modifiers.put(ModifiersTG.AM_DISTANCE,"1000,2000,3000");

**Comment by fteel** [comment]
Hey Michael,



Thanks for the quick response.  Let me dig a little deeper into what you’ve suggested and see what I can come up with.



V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by fteel** [comment]
Hi Michael,



1.      We have, RendererSettings.getInstance().setSymbologyStandard( RendererSettings.Symbology_2525C );
2.      Task->Disrupt

The image on the right is from 2525C.  It depicts the symbol with the longest arrow on the bottom and has the letter “D” on the middle arrow.  The placement image on the right shows the longest arrow on the top side and there is no “D”.  While this render could be rotated to be in proper position it is expected that it will render as depicted in 2525C with the longest arrow on the bottom and the “D” on the middle arrow.





3.      Mobility/Survivability->Obstacles->Obstacles Effect-> Disrupt

The image on the right is from 2525C.  It depicts the symbol with the longest arrow on the top.  There is no way even with rotating the image that the render on the right can be aligned with the image from 2525C on the left.



4.      The issues with Dummy Minefield, Dynamic Depiction, Contaminated Areas are that they are being rendered as triangles.
5.      I am looking into the Rectangular images to see if we are passing in the modifiers properly.





V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by fteel** [comment]
Hi Michael,



It appears that although we set the modifiers and the values actually represent a rectangle the values are not disparate enough to make the image appear as a rectangle to the user.



V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by fteel** [comment]
Michael,



I am having trouble making sense out of modifiers.put(ModifiersTG.AM_DISTANCE,"1000,2000,3000");



The best I can do is get an elongated rectangle that is higher than wider.  The numbers don’t seem to have the same effect on the areas NFA ….. as they do on a Rectangular Target.



Can you help me out? Is there any documentation?



V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by michael-spinelli** [comment]
It's all in the MilStd 2525

No Fire Area has 2 points and the width is determined by 1 "AM" value
modifiers.put(ModifiersTG.AM_DISTANCE,"1000");
![image](https://user-images.githubusercontent.com/5972575/94046876-4061cc80-fd9f-11ea-89ca-33ddf72d92d0.png)


Rectangular Target has 1 point (center) with 1 "AN" value for attitude/angle and 2 "AM" values for length and width
modifiers.put(ModifiersTG.AN_AZIMUTH,"45");//If I remember correctly, it can be degrees or mils depending on the symbol.  Check MilStd.
modifiers.put(ModifiersTG.AM_DISTANCE,"1000, 2000");//in meters, AM, AM1, AM2....
![image](https://user-images.githubusercontent.com/5972575/94046230-6fc40980-fd9e-11ea-8f98-3d8abaf7359a.png)

**Comment by michael-spinelli** [comment]
Also [getSymbolDef](https://github.com/missioncommand/mil-sym-android/wiki#52getsymboldef) will help you figure out what modifiers each symbol has.

**Comment by michael-spinelli** [comment]
For your areas that are coming out as triangle, are you saying you're adding more than three set of coordinates and they're still rendering as a triangle?

**Comment by fteel** [comment]
So for instance the Dummy Minefield (dynamic)  we are passing in



-87.6886362173281,30.381411101605323

-87.6886362173281,30.382169886229793

-87.68938992380288,30.381790493917556



And getting this









V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by michael-spinelli** [comment]
The picture didn't show up.  
Can you reply directly on the [issue tracker in github](https://github.com/missioncommand/mil-sym-android/issues/96) rather than using email?

Also, I plotted those points and they make a triangle.

**Comment by fteel** [comment]
![image](https://user-images.githubusercontent.com/62301296/94115078-da149280-fe0e-11ea-8aed-8a266ce000ae.png)

**Comment by michael-spinelli** [comment]
I plotted the points from your email.  It looks fine to me.  What am I missing?:
![point2](https://user-images.githubusercontent.com/5972575/94144311-f9c3af00-fe3e-11ea-816b-7e1c17eb2f74.png)

I flipped the lat and lon values and got this as well:
![point1](https://user-images.githubusercontent.com/5972575/94144453-237cd600-fe3f-11ea-91cd-0923acaed0c3.png)

**Comment by fteel** [comment]
So I sent the coordinates as a reference along with the picture that’s attached. The issue is that the boundary of the area as depicted in 2525C is not a triangle.





V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by michael-spinelli** [comment]
It's a polygon based area.   It works similar to Fire Support Area (FSA) and it takes the shape of whatever points you give it.

**Comment by fteel** [comment]
Michael,



If I wanted to have symbol say like a hexagon is there an order in which the points have to be entered in the method RenderSymbolAsMilStdSymbol?









V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by michael-spinelli** [comment]
So, using something like a general area?
For something simple like that, I don't believe order matters.

You could also use the code "BS_AREA--------" for basic shape area.  It's not a milstd symbol so you can't pass it to another system but you can draw a simple area with it.

**Comment by fteel** [comment]
Yes like a general area.  For some reason we use a triangle for every general area  that does not have a specific shape.  Which make is quite confusing if you learn the symbols from the F.M. and then see a triangle.  I don’t think I’ll be the one to fix them all but I need to understand how it works and at least have a couple examples in code.





Is there an example that you could show me for a general area and how to use “BS_AREA--------” …. Would that be the sidc?





V/R,



Forrest G Teel Jr

Software Engineer IV

Augustine Consulting Inc

Cell: 339-236-1472

Office: 774-413-5388

fteel@aciedge.com<mailto:fteel@aciedge.com>



Winning is in the preperation not in the game!











You are hereby notified that I DO NOT have the authority to direct you in any way to alter your contractual obligation. Further, if the Government, as a result of the information obtained from this email discussion DOES desire to alter your requirements, changes will be issued in writing and signed by the contracting officer. You should take no action on any change unless you receive such a contract modification.

**Comment by michael-spinelli** [comment]
Yes, "BS_AREA--------" is just an SIDC we made up for the purpose of drawing a basic shape area.  Like the other areas, it requires a minimum of 3 points but can have as many as you want to make whatever shape you want.  They're no different in usage from the other areas you've been drawing (minefields, NFA, etc...).
